core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/CheckBoxEditorDirective.js

```diff
@@ -0,0 +1,291 @@
+ */
+
+// NOTE TO DEVELOPERS: This is meant to be a reusable directive, so its only
+// dependencies should be standard utility services. It should not have any
+// concept of "state in an exploration".
+oppia.directive('checkBoxEditor', [
+  'QuestionIdService', 'AnswerGroupObjectFactory', 'RuleObjectFactory',
+  'StatusObjectFactory',
+  function(QuestionIdService, AnswerGroupObjectFactory, RuleObjectFactory,
+      StatusObjectFactory) {
+    return {
+      restrict: 'E',
+      scope: {
+        // A unique ID that allows events to be broadcast specifically to this
+        // directive.
+        getUniqueId: '&uniqueId',
+        getCustomizationArgs: '&customizationArgs',
+        getAnswerGroups: '&answerGroups',
+        getRawDefaultOutcome: '&defaultOutcome',
+        saveCustomizationArgs: '&',
+        saveAnswerGroups: '&',
+        saveDefaultOutcome: '&',
+        addState: '&'
+      },
+      templateUrl: 'simpleEditorQuestions/ItemSelectionInput',
+      controller: [
+        '$scope', '$timeout', 'alertsService',
+        function($scope, $timeout, alertsService) {
+          // Note that a questionId generated in this way may contain spaces,
+          // since it is just the state name.
+          $scope.questionId = $scope.getUniqueId();
+
+          $scope.getSubfieldId = function(label) {
+            return QuestionIdService.getSubfieldId($scope.questionId, label);
+          };
+
+          $scope.getFieldId = function(index) {
+            return $scope.questionId + '.' + index;
+          };
+
+          var openChoiceEditor = function(index) {
+            $scope.$broadcast('openEditorHtmlField', {
+              fieldId: $scope.getFieldId(index)
+            });
+          };
+
+          $scope.$on('newInteractionIdSelected', function(evt, data) {
+            if (data.targetId === $scope.getUniqueId()) {
+              openChoiceEditor(0);
+            }
+          });
+
+          $scope.getChoices = function() {
+            return $scope.getCustomizationArgs().choices.value;
+          };
+
+          $scope.getDefaultOutcome = function() {
+            var defaultOutcome = $scope.getRawDefaultOutcome();
+            if (defaultOutcome.feedback.length === 0) {
+              defaultOutcome.feedback.push('');
+            }
+            return defaultOutcome;
+          };
+
+          $scope.isCorrectAnswer = function(value) {
+            var answerGroups = $scope.getAnswerGroups();
+            if(answerGroups[0]["rules"][0]["inputs"]["x"].includes(value) ){
+              return true;
+             }
+             else{
+               return false;
+             }
+
+          };
+
+          $scope.addChoice = function() {
+            var newCustomizationArgs = $scope.getCustomizationArgs();
+            var choiceNames = newCustomizationArgs.choices.value;
+
+            // If there is a currently-empty option, do nothing and focus into
```

Comment:
This says that you're going to focus into the field, but if I understand the code correctly, you're actually returning an error instead. Which is correct?